lib/AST/GenericSignature.cpp

```diff
@@ -605,6 +605,10 @@ SmallVector<Requirement, 4> GenericSignatureImpl::requirementsNotSatisfiedBy(
     return result;
   }
 
+  // If the canonical signatures match, all requirements are satisfied.
+  if (this->getCanonicalSignature() == otherSig->getCanonicalSignature())
```

Comment:
This is good to have independent of this patch.

Reply:
FYI, the exact same changes around `requirementsNotSatisfiedBy` are also part of #30712